pkg/build/api/v1/types.go

```diff
@@ -371,6 +375,13 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// RunBash holds a bash script to be executed in a container running the
+	// build output image.
+	RunBash string `json:"runBash,omitempty" description:"bash script to be executed in a container running the build output image"`
```

Comment:
>  Is it a problem, do we need trackability of what is the command to run tests?

Doesn't strike me as terribly critical, these are unit tests, not acceptance tests, and they run after the image is committed meaning they can't manipulate the image that's getting pushed to the registry.  So it really doesn't matter that much what command is run here.

Also the command that was run post-commit will be in the build logs (well it better be, @rhcarvalho please ensure it is, should be an Info message (ie not requiring verbose log levels), but i assume it is).

if you want it auditable, you can put the logic in your assemble script.

As for the api, I like:
```
{
  "postCommit": {
    "entrypoint: ["bash", "-c"],
    "cmd": ["rake test"]
  }
}
```
With "entrypoint" being optional and defaulting to "bash -c", meaning for our s2i images you could just specify:
```
{
  "postCommit": {
    "cmd": ["rake test"]
  }
}
```

That mimics the docker entrypoint/command api.  The only point of confusion is that the "default" entrypoint does not match the entrypoint specified in the image.  Suggestions for how to deal with that are welcome..

Alternatively we can copy the Container api:
```
	// Optional: The docker image's entrypoint is used if this is not provided; cannot be updated.
	// Variable references $(VAR_NAME) are expanded using the container's environment.  If a variable
	// cannot be resolved, the reference in the input string will be unchanged.  The $(VAR_NAME) syntax
	// can be escaped with a double $$, ie: $$(VAR_NAME).  Escaped references will never be expanded,
	// regardless of whether the variable exists or not.
	Command []string `json:"command,omitempty"`
	// Optional: The docker image's cmd is used if this is not provided; cannot be updated.
	// Variable references $(VAR_NAME) are expanded using the container's environment.  If a variable
	// cannot be resolved, the reference in the input string will be unchanged.  The $(VAR_NAME) syntax
	// can be escaped with a double $$, ie: $$(VAR_NAME).  Escaped references will never be expanded,
	// regardless of whether the variable exists or not.
	Args []string `json:"args,omitempty"`
```

with the difference being that for "Command" if no value is provided, we do not use the docker image's entrypoint, we use "bash -c".